### DISCUSSION TITLE
Github API can't update required status checks of protected branch

### DISCUSSION THREAD

**Discussion by tonyzwei** [community/community#24424] [Apps, API and Webhooks]
<p>I am an owner of my org repositories and I have more than a thousand of it, so I write automation scripts with API calls to manage GitHub repository settings. The following is an example of the issue:</p>
<p>I have a github repository which has branch protection settings for relase branch:</p>
<p>{<br>
“url”: “<a href="https://api.github.com/repos/mdsol/go-cd-test/branches/release/protection" rel="nofollow noopener">https://api.github.com/repos/mdsol/go-cd-test/branches/release/protection</a>”,<br>
“required_pull_request_reviews”: {<br>
“url”: “<a href="https://api.github.com/repos/mdsol/go-cd-test/branches/release/protection/required_pull_request_reviews" rel="nofollow noopener">https://api.github.com/repos/mdsol/go-cd-test/branches/release/protection/required_pull_request_reviews</a>”,<br>
“dismiss_stale_reviews”: false,<br>
“require_code_owner_reviews”: false,<br>
“required_approving_review_count”: 1<br>
},<br>
“enforce_admins”: {<br>
“url”: “<a href="https://api.github.com/repos/mdsol/go-cd-test/branches/release/protection/enforce_admins" rel="nofollow noopener">https://api.github.com/repos/mdsol/go-cd-test/branches/release/protection/enforce_admins</a>”,<br>
“enabled”: false<br>
}<br>
}</p>
<p>I ran the following command to update required status checks of protected branch:</p>
<p>curl -sS -X PATCH “$GITHUB_API/repos/mdsol/go-cd-test/branches/release/protection/required_status_checks” -H “Authorization: token $TRAVIS_TOKEN” -H “Accept: application/vnd.github.luke-cage-preview+json” -d ‘{“strict”:false,“contexts”:[“GoCD”]}’</p>
<p>but it returned:<br>
{<br>
“message”: “Required status checks not enabled”,<br>
“documentation_url”: “<a href="https://developer.github.com/v3/repos/branches/#update-required-status-checks-of-protected-branch" rel="nofollow noopener">https://developer.github.com/v3/repos/branches/#update-required-status-checks-of-protected-branch</a>”</p>
<p>}</p>
<p>I searched the documents and can’t find anything helpful to enable “Required status checks” through API calls. Do you have any ideas and suggestions? Thanks!</p>
<p>I don’t want to use “-X PUT” because I want to keep existing branch protection rules and they are very diverse in repositories.</p>

**Comment by AndreaGriffiths11** [comment]
<p>Hi <a class="mention" href="/tonyzwei">@tonyzwei</a>,</p>
<p>This post was moved to a different board that fits your topic of discussion a bit better. This means you’ll get better engagement on your post, and it keeps our community organized so users can more easily find information.</p>
<p>As you’ll notice, your topic is now here in the GitHub API board. No action is needed on your part; you can continue the conversation as normal here.</p>
<p>Cheers!</p>

**Comment by frconil** [comment, marked as answer]
<p>Hi Tony,</p>
<p>As you found out, you will need to enable the required status checks via <a href="https://developer.github.com/v3/repos/branches/#update-branch-protection" rel="nofollow noopener">the Update Branch Protection API Endpoint</a> first.</p>
<p>You will have to extract the current configuration first with the <a href="https://developer.github.com/v3/repos/branches/#get-branch-protection" rel="nofollow noopener">Get Branch Protection API Endpoint</a>.</p>
<p>François</p>
